### DISCUSSION TITLE
passing Arrays in params in useFetch (and $fetch)

### DISCUSSION THREAD

**Discussion by JensErikChristiansen** [nuxt/nuxt#17404] [Questions] [answered]
Hi! I'm trying to pass an array as a property in `params` in `useFetch` (and `$fetch`, which I assume take the same options). Something like:

```javascript
const { data } = useFetch('/api/cars', {
  params: {
    makes: ['Honda', 'Mazda', 'Dodge']
  }
}
```

But when it gets to the server, it seems that `makes` gets stringified as `[object Object]`. Is there a way to send `makes` without having to stringify the array?

Thank you!

Jens

**Comment by Diizzayy** [comment, marked as answer]
@JensErikChristiansen You can utilize `useQuery` in your api route as seen below to properly extract the params

```ts
export default defineEventHandler((event) => {
  const params = useQuery(event)

  // /api/demo?makes=Honda&makes=Mazda&makes=Dodge
  console.log('URL', event.req.url )

  // params { makes: [ 'Honda', 'Mazda', 'Dodge' ] }
  console.log('params', params)

  return params?.makes || []
})
```

**Reply by JensErikChristiansen** [reply]
yeah, unfortunately when doing `params = useQuery(event)`, params.makes becomes '[object Object]'

**Reply by JensErikChristiansen** [reply]
Ah, sorry, I had simplified my example too much. The true structure of the data that I'm sending back is:

```javascript
useFetch("/api/vehicles", {
  params: {
    makes: [
      {
        makeName: "BMW",
        modelNamesList: ["3 Series"],
      },
      {
        makeName: "Honda",
        modelNamesList: ["Accord", "Civic"],
      },
    ],
  },
});

```

`makes` is indeed an array, but each element inside of it is just [object Object].
```
makes: [ '[object Object]', '[object Object]' ]
```

**Reply by Diizzayy** [reply]
> Ah, sorry, I had simplified my example too much. The true structure of the data that I'm sending back is:

@JensErikChristiansen Ahh, that explains it. Your objects must be serialized in some way before passing them as parameters, perhaps by using `JSON.stringify`

As you know url queries are strings and passed as `/example?makes=`, hence the need for serializing.

```ts
useFetch('/api/demo', {
  params: {
    makes: JSON.stringify([
      {
        makeName: "BMW",
        modelNamesList: ["3 Series"],
      },
      {
        makeName: "Honda",
        modelNamesList: ["Accord", "Civic"],
      },
    ])
  }
})
```

PS: @JensErikChristiansen I wouldn't advise using query string parameters for any advanced data structures

**Reply by JensErikChristiansen** [reply]
thank you, @Diizzayy! Yeah I figured as such. I've thought about maybe making it a POST request and passing params through the body, but I'm only getting data, not creating. Any other ideas besides redesigning my endpoint?

**Reply by Diizzayy** [reply]
@JensErikChristiansen I understand your hesitancy about using `POST`, but I think it is in fact the best direction here

**Reply by JensErikChristiansen** [reply]
Thanks again!
